### PR TITLE
Feature/mp 2429 i os integrate qr code reader in the demo app

### PR DESCRIPTION
**Pull request by danial-rokt**
### Background ###

Need to provide an easy way to preview the DCUI Layout in iOS app. The plan is to be able to scan the QR code on the layout page on OP and render the QR code in the Demo application.

Fixes [issue](https://rokt.atlassian.net/browse/MP-2429), [issue](https://rokt.atlassian.net/browse/MP-2431) , [issue](https://rokt.atlassian.net/browse/MP-2433) 

### What Has Changed: ###

* Updated to version `4.0.0-alpha.3`
* New page `Layout library` implemented
* `CodeScanner` library integrated to scan QR code in SwiftUI
* Camera permission added to the Demo application
* Init data is extracted from QR code and `Rokt.initWith` method is called after the scan
* Execute data is extracted from QR code and `Rokt.execute` method is called after 3 seconds to preview the layouts
* Refresh preview button  
* Error page implemented on unsupported QR code

### How Has This Been Tested? ###

Tested locally:

The video shows 3 barcodes in the simulator:
1: BottomSheet
2: Embedded
3: Invalid QR code

https://github.com/ROKT/rokt-demo-ios/assets/49017604/eb7bbfeb-0ff6-4cf6-8aa8-46fba2d43b76

The video that shows the functionality on the phone:

https://github.com/ROKT/rokt-demo-ios/assets/49017604/07ffc6fe-dd47-4a88-89b2-a56625a261d6

### Notes

`CodeScanner` library has functionality to bring in imagepicker while testing on the simulator

### Checklist: ###

- [x] I have performed a self-review of my own code.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have verified that CI completes successfully.
- [x] All insignificant commits have been squashed.

### CODE REVIEW

**Review by emmanuel-rokt — APPROVED** [review]

File: RoktDemo.xcworkspace/xcuserdata/danial.xcuserdatad/xcdebugger/Breakpoints_v2.xcbkptlist
Comment (emmanuel-rokt) [inline RoktDemo.xcworkspace/xcuserdata/danial.xcuserdatad/xcdebugger/Breakpoints_v2.xcbkptlist:1]: do we need this in source control? it's a debugger file

Comment (danial-rokt) [reply RoktDemo.xcworkspace/xcuserdata/danial.xcuserdatad/xcdebugger/Breakpoints_v2.xcbkptlist:1]: no, good point. removed.

File: RoktDemo/Info.plist [hunk 1]
```diff
@@ -22,6 +22,8 @@
 	<string>$(CURRENT_PROJECT_VERSION)</string>
 	<key>LSRequiresIPhoneOS</key>
 	<true/>
+	<key>NSCameraUsageDescription</key>
+	<string>Need the permission to scan the QR code.</string>
```
Comment (emmanuel-rokt) [inline RoktDemo/Info.plist:26]: nit: Needed to scan QR codes

**Comment by emmanuel-rokt** [comment]
just small comments that aren't merge blockers. approving in case those won't be changed. Thanks for the effort

**Review by thomson-t — APPROVED** [review]

File: RoktDemo/Info.plist (hunk 1 above)
Comment (danial-rokt) [reply RoktDemo/Info.plist]: addressed

**Pull request merged**